gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/upgrade/upgrader/environment/MissingEnvironmentUpgrader.java

```diff
@@ -45,45 +51,43 @@ public boolean upgrade() {
     private void updateEnvironmentFromApi(final Set<Boolean> upgradeStatus) {
         var projection = Projections.fields(Projections.include("_id", "environmentId"));
         var bulkActions = new ArrayList<UpdateManyModel<Document>>();
-        template
-            .getCollection("apis")
+        this.getCollection("apis")
             .find()
             .projection(projection)
             .forEach(api ->
                 bulkActions.add(
                     new UpdateManyModel<>(
-                        new Document("api", api.getString("_id")),
-                        new Document("$set", new Document("environmentId", api.getString("environmentId")))
+                        Filters.and(Filters.eq("api", api.getString("_id")), Filters.exists("environmentId", false)),
+                        Updates.set("environmentId", api.getString("environmentId"))
                     )
                 )
             );
 
         if (!bulkActions.isEmpty()) {
             // This upgrade is only done on data created before 3.17.0 as ApiKey#api as been deprecated
-            upgradeStatus.add(template.getCollection("keys").bulkWrite(bulkActions).wasAcknowledged());
-            upgradeStatus.add(template.getCollection("plans").bulkWrite(bulkActions).wasAcknowledged());
-            upgradeStatus.add(template.getCollection("subscriptions").bulkWrite(bulkActions).wasAcknowledged());
+            upgradeStatus.add(this.getCollection("keys").bulkWrite(bulkActions).wasAcknowledged());
+            upgradeStatus.add(this.getCollection("plans").bulkWrite(bulkActions).wasAcknowledged());
+            upgradeStatus.add(this.getCollection("subscriptions").bulkWrite(bulkActions).wasAcknowledged());
         }
     }
 
     private void updateEnvironmentFromSubscription(final Set<Boolean> upgradeStatus) {
         var projection = Projections.fields(Projections.include("_id", "environmentId"));
         var bulkActions = new ArrayList<UpdateManyModel<Document>>();
-        template
-            .getCollection("subscriptions")
+        this.getCollection("subscriptions")
             .find()
             .projection(projection)
             .forEach(subscription ->
                 bulkActions.add(
                     new UpdateManyModel<>(
-                        Filters.in("subscriptions", subscription.getString("_id")),
-                        new Document("$set", new Document("environmentId", subscription.getString("environmentId")))
+                        Filters.and(Filters.eq("subscriptions", subscription.getString("_id")), Filters.exists("environmentId", false)),
+                        Updates.set("environmentId", subscription.getString("environmentId"))
```

Comment:
Don't you miss the exist filter?

Reply:
it is here
`, Filters.exists("environmentId", false)`